worldedit-core/src/main/java/com/boydti/fawe/beta/implementation/processors/HeightmapProcessor.java

```diff
@@ -0,0 +1,108 @@
+package com.boydti.fawe.beta.implementation.processors;
+
+import com.boydti.fawe.beta.IBatchProcessor;
+import com.boydti.fawe.beta.IChunk;
+import com.boydti.fawe.beta.IChunkGet;
+import com.boydti.fawe.beta.IChunkSet;
+import com.boydti.fawe.beta.implementation.blocks.CharBlocks;
+import com.boydti.fawe.beta.implementation.lighting.HeightMapType;
+import com.sk89q.worldedit.extent.Extent;
+import com.sk89q.worldedit.world.World;
+import com.sk89q.worldedit.world.block.BlockState;
+import com.sk89q.worldedit.world.block.BlockType;
+import com.sk89q.worldedit.world.block.BlockTypes;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.BitSet;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+
+public class HeightmapProcessor implements IBatchProcessor {
+    private static final HeightMapType[] TYPES = HeightMapType.values();
+    private static final BlockType RESERVED = BlockTypes.__RESERVED__;
+    private static final int SECTION_SIDE_LENGTH = 16;
+    private static final int BLOCKS_PER_Y_LEVEL = SECTION_SIDE_LENGTH * SECTION_SIDE_LENGTH;
+
+    private final World world;
+
+    public HeightmapProcessor(World world) {
+        this.world = world;
+    }
+
+    @Override
+    public IChunkSet processSet(IChunk chunk, IChunkGet get, IChunkSet set) {
```

Comment:
Might be nice to have the option to add this as a post-processor as well. For a lot of people, heightmaps don't matter too much (typically build teams etc. who value speed over vanilla behaviour). This would match with having a config option to disable HeightMap processing altogether.

Reply:
> Might be nice to have the option to add this as a post-processor as well. For a lot of people, heightmaps don't matter too much (typically build teams etc. who value speed over vanilla behaviour). This would match with having a config option to disable HeightMap processing altogether.

I disagree, heightmaps affect a lot, such as maps, mob spawning, rainfall and perhaps even more.
From my testing, the heightmap updates have no noticable effect on speed.

Reply:
You might be surprised as to the effect it can have for brush size 10-30. Typically smoothing brushes in this case. One of the main points of FAWE is high customisation, so by default we have the "better" behaviour, but allow people to customise to their specific use-case.

Reply:
Well if the heightmap updates are enabled by default and the effort to implement that option is not too high, as a user I am fine with it.
However believe me that I have been testing brushes as well, I even used Timers to increase my click speed xD

Reply:
Not sure if it's worth to have it as post processor (altough I guess it can just run the same code?). Disabling it might make sense if you only want to create a schematic from your edits, but as even third party plugins can depend on heightmaps, it should be discouraged for public servers.

Reply:
I've been considering adding lighting as a true post-process (currently send-before-lighting still waits for lighting). It's not really required for this PR, and could be included when I get round to doing the aforementioned lighting shizzle

Reply:
well first it is good if it works properly, and from what I can tell, you guys are almost there :D